packages/circuit-ui/components/NotificationToast/NotificationToast.docs.mdx

```diff
@@ -0,0 +1,137 @@
+
+### Notify
+
+Use 'Notify' variant for a warning messages, information to be aware of.
+
+### Alert
+
+Use 'Alert' variant for error messages.
+
+## Usage in code
+
+First, wrap your application in the `ToastProvider` which keeps track of the open toasts and ensures the accessibility of the toast.
+
+```tsx
+// _app.tsx for Next.js or App.js for CRA
+import { ToastProvider, BaseStyles } from '@sumup/circuit-ui';
+
+export default function App() {
+  return (
+      <BaseStyles />
+      <ToastProvider>{/* Your content goes here... */}</ToastProvider>
+  );
+}
+```
+
+Then, use the `useNotificationToast` hook to open a toast from a component:
+
+```tsx
+import { useNotificationToast, Button, Body } from '@sumup/circuit-ui';
+
+export function App({ name }) {
+  const { setToast } = useNotificationToast();
+
+  const handleClick = () => {
+    setToast({
+      body: 'This is a toast message',
+      closeButtonLabel: 'Close modal',
+    });
+  };
+
+  return <Button onClick={handleClick}>Open toast</Button>;
+}
+```
+
+## Accessibility
+
+By their nature as small quick messages, or interruptions, toasts map to [live regions](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions),
+and screen readers are gonna automatically announce the raw text within without any need to move the user's focus.
+Live region must always exist on the page at render, before the toast is generated.
+To ensure that the toast is always announced as a whole, `aria-atomic` should be set to `"true"` (default value is `"false"`).
+
+### Role
+
+The `aria-role` can be adapted based on the context.
+For the most generic use cases that provides an updated status, the role can be set to `"status"`. It has a `aria-live="polite"` meaning the screen reader will wait before announcing the update.
+If it is an important message, typically time-sensitive, like an error, use `role="alert"`. It has `aria-live="assertive"` which means it will interrupt whatever the user is doing.
+The third role that can be given to the toast messages is `"log"`. It should be used when the order of the messages matters. It has a `aria-live="polite"`.
+
+Since the screen readers are gonna announce the raw text within toast messages, without structure, keyboard users won’t be able to move to the interactive elements with a clear path.
+For example, there is a toast message with the text “You successfully updated your data.”, followed by a link or button with the text "Back".
+Screen readers are gonna communicate a message as “You successfully updated your data. Back.”, meaning that the semantics of the link "Back" is not announced.
+People using a screen reader may think that the "Back" is an interactive element, but they will not know which element to search for.
+In this case you will need a modal or dialog with a `role="dialog"`.
+Ideally, the toast messages should not contain any interactive elements.
+
+### Best practices
```

Comment:
Although all of the best practices below do apply to our toast, none of them are actionable by developers. We define the color, the positioning, etc.

So similar to what Connor mentioned above, I'd either remove it altogether (to keep developers focused on what they really have an impact on) or make it very clear that this is Circuit's concern, not theirs

(☝️ and we should do the same in all components.)

Reply:
I'm tending towards removing it, because many of these guidelines would have to be repeated in basically every component (e.g. color contrast).

This is rather an accessibility statement to me: at some point we'll state somewhere that we are WCAG AA compliant, which implies that we must meet all AA success criteria, including color, focus for interactive elements, etc.

But would like to hear your thoughts about it 🙂

Reply:
I really like these sections and would like to keep them. This information is very helpful for maintainers, makes our accessibility efforts more visible, and can inspire and educate developers on these principles.

You're right that we should make this context clear. What about one of these statements at the start of the section?

> This component was created with the following accessibility principles in mind:

> This component adheres to the following principles from the 
Web Content Accessibility Guidelines (WCAG):

Reply:
I was just going through these again, and I am more towards keeping it too. I will make it more clear and update it. I like your suggestion Connor for the start.